### PR TITLE
Refactoring navigation and fixing rendering bugs

### PR DESCRIPTION
**Pull request by Thomas-Gallant**
Changelog:
- setting defaultSource in Image to eliminate bug where image visibly flashes when rerendering
    - did this for all images we have
- Removing nested navigation to improve performance as recommended by react navigation docs
    - https://reactnavigation.org/docs/nesting-navigators/#best-practices-when-nesting
    - This fixes the issue where user is stuck in a lesson / drill / game when trying to navigate to the root lesson page or drill page or play page until a refresh happens. (Stack navigators removed)
- Adding "unmountOnBlur" setting so all pages unmount when navigating away. This seems to have greatly improved performance of playwright testing (20 seconds per test instead of 40 on my local machine. This also fixed the issue of home page buttons jittery resizing (header height no longer goes to 0 when navigating to different pages)
- Contact page tests have more accurate locators instead of using `last()` locators
- Removing all `last()` locators from playwright tests as they are no longer needed.
- Reducing code duplication in Header component

## Checklist for completing pull request:
- [ ] Test functionality on Mobile, Web, and Desktop
- [x] Verify that any tests for new functionality are created and functional.
- [x] If needed, update the Readme

## Summary by CodeRabbit

- **New Features**
	- Introduced `defaultSource` property in image components to enhance loading behavior and user experience.
	- Updated navigation logic with more specific page identifiers (e.g., "LandingPage", "ProfilePage") for clarity.
	- Improved handling of button interactions in navigation components for better responsiveness.

- **Bug Fixes**
	- Resolved flickering issues in image components by implementing default image sources.

- **Documentation**
	- Enhanced comments in the code to clarify the behavior of hooks and components.

- **Refactor**
	- Streamlined navigation logic and component rendering for improved maintainability and readability.

### CODE REVIEW

**Comment by Thomas-Gallant** [comment]
@coderabbitai review

File: e2e/web/specs/play-sudoku.spec.ts [hunk 1]
```diff
@@ -68,7 +68,7 @@ test.describe("complete game", () => {
     const endGameModal = new EndGameModalComponent(resumeGame);
     await endGameModal.newGame.click();
     const header = new HeaderComponent(resumeGame);
-    await header.statistics.last().click(); // todo: stop using last (fix infinite stack)
+    await header.statistics.click(); // todo: stop using last (fix infinite stack)
```
Comment (Gregory711) [inline e2e/web/specs/play-sudoku.spec.ts:71]: can get rid of the todo comment here

**Comment by Gregory711** [comment]
https://github.com/Sudokuru/Frontend/blob/refactoringNavigation/cypress/e2e/drills.cy.ts#L72
can remove this last as well

File: e2e/web/specs/play-sudoku.spec.ts (hunk 1 above)
Comment (Thomas-Gallant) [reply e2e/web/specs/play-sudoku.spec.ts:71]: Good call!

**Review by Gregory711 — APPROVED** [review]

**Pull request merged**